src/api/controllers/db-controller.ts

```diff
@@ -211,6 +215,65 @@ export function parseDbEvent(dbEvent: DbEvent): TransactionEvent {
   }
 }
 
+/**
+ * Fetch block from datastore by blockHash or blockHeight (index)
+ * If both blockHeight and blockHash are provided, blockHeight is used.
+ * If neither argument is present, the most recent block is returned.
+ * @param db -- datastore
+ * @param blockHash -- hexadecimal hash string
+ * @param blockHeight -- number
+ */
+export async function getRosettaBlockFromDataStore(
+  db: DataStore,
+  blockHash?: string,
+  blockHeight?: number
+): Promise<{ found: true; result: RosettaBlock } | { found: false }> {
```

Comment:
You can use the `FoundOrNot<T>` generic type here

```suggestion
): Promise<FoundOrNot<RosettaBlock>> {
```

Reply:
There are several places where `FoundOrNot` can be used; we'll try to get them done on Monday.